source/common/api/BUILD

```diff
@@ -22,22 +22,27 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "os_sys_calls_lib",
-    srcs = ["os_sys_calls_impl.cc"] + select({
-        "//bazel:linux_x86_64": ["os_sys_calls_impl_linux.cc"],
-        "//bazel:linux_aarch64": ["os_sys_calls_impl_linux.cc"],
-        "//bazel:linux_ppc": ["os_sys_calls_impl_linux.cc"],
-        "//bazel:linux_mips64": ["os_sys_calls_impl_linux.cc"],
-        "//conditions:default": [],
-    }) + envoy_select_hot_restart(["os_sys_calls_impl_hot_restart.cc"]),
-    hdrs = ["os_sys_calls_impl.h"] + select({
-        "//bazel:linux_x86_64": ["os_sys_calls_impl_linux.h"],
-        "//bazel:linux_aarch64": ["os_sys_calls_impl_linux.h"],
-        "//bazel:linux_ppc": ["os_sys_calls_impl_linux.h"],
-        "//bazel:linux_mips64": ["os_sys_calls_impl_linux.h"],
-        "//conditions:default": [],
-    }) + envoy_select_hot_restart(["os_sys_calls_impl_hot_restart.h"]),
+    srcs = select({
+        "//bazel:linux": [
+            "posix/os_sys_calls_impl.cc",
+            "posix/os_sys_calls_impl_linux.cc",
+        ],
+        "//conditions:default": ["posix/os_sys_calls_impl.cc"],
+    }) + envoy_select_hot_restart(["posix/os_sys_calls_impl_hot_restart.cc"]),
+    hdrs = select({
+        "//bazel:linux": [
+            "posix/os_sys_calls_impl.h",
+            "posix/os_sys_calls_impl_linux.h",
+        ],
+        "//conditions:default": ["posix/os_sys_calls_impl.h"],
+    }) + envoy_select_hot_restart(["posix/os_sys_calls_impl_hot_restart.h"]),
+    strip_include_prefix = select({
+        "//conditions:default": "posix",
+    }),
     deps = [
         "//include/envoy/api:os_sys_calls_interface",
         "//source/common/singleton:threadsafe_singleton",
-    ],
+    ] + select({
+        "//conditions:default": [],
```

Comment:
This can be deleted, right?

Reply:
We'd prefer to leave it (fill-in-the-gap adds windows specifics there in a upcoming patch, if that's alright.) Trying to keep as many changes in the upcoming commits to be line-oriented diffs.

Reply:
OK sounds good.

Reply:
we found that we didn't need the subsequent changes in the later commit for Windows, so we will remove this bit

Reply:
We just re-reviewed and our modification turned out to be unnecessary. Follow-up commit removes this select() as requested (and cleans up win32 deps from the follow-on patch).